src/components/AppMap.vue

```diff
@@ -6,36 +6,221 @@ import GoogleMapsApiLoader from "google-maps-api-loader";
 
 import { getDeviceGeoLocation } from "../utils";
 import styles from "../style/google-maps";
-
-export const BOUNDARIES_NETHERLANDS = { north: 53.53, south: 50.74, west: 3.35, east: 7.25 };
+import { mapGetters, mapState, mapActions } from "vuex";
 
 export default {
   data() {
     return {
       google: null,
-      map: null
+      map: null,
+      fullCoverage: null,
+      areaCoverages: {},
+      originMarkers: [],
+      intersectionPaths: []
     };
   },
+  computed: {
+    ...mapGetters("locations", ["getLocationTypeByValue"]),
+    ...mapGetters("ranges", ["rangesWithOrigin"]),
+    ...mapState("areas", ["mapBoundaries", "areas"]),
+    ...mapState("ranges", {
+      activeRangeId: state => state.activeId
+    }),
+
+    maskPath: function() {
+      return [
+        { lat: this.mapBoundaries.north + 5, lng: this.mapBoundaries.west - 5 },
+        { lat: this.mapBoundaries.north + 5, lng: this.mapBoundaries.east + 5 },
+        { lat: this.mapBoundaries.south - 5, lng: this.mapBoundaries.east + 5 },
+        { lat: this.mapBoundaries.south - 5, lng: this.mapBoundaries.west - 5 }
+      ];
+    }
+  },
+  watch: {
+    areas: function() {
+      this.drawOrigins();
+      this.drawCoverage();
+    },
+
+    activeRangeId: function(activeRangeId) {
+      for (let rangeId in this.areaCoverages) {
```

Comment:
Avoid using for. Go for imutable constructions like map, filter, reduce

Reply:
@janjaap Since range changes with every iteration, I thought using const would crash the construction. Turns out it doesn't crash. How is this? Is it because it's treated at the block level, in every iteration?
The hasOwnProperty check is already there, it's just not part of this code snippet. You can see it if you expand the code.

@aditudorache this is an object hash, not an array. I could refactor it to an array, but I don't really see to what benefit...